lib/spack/spack/test/file_list.py

```diff
@@ -36,12 +36,13 @@
 @pytest.fixture()
 def library_list():
     """Returns an instance of LibraryList."""
+    # Test all valid extensions: ['.a', '.dylib', '.so']
     l = [
         '/dir1/liblapack.a',
-        '/dir2/libfoo.dylib',
+        '/dir2/libpython3.6.dylib',  # name may contain periods
         '/dir1/libblas.a',
-        '/dir3/libbar.so',
-        'libbaz.so'
+        '/dir3/libz.so',
+        'libmpi.so.20.10.1',  # shared object libraries may be versioned
```

Comment:
No, that should still work I think. I mean, most of the time, there is a `libmpi.so` that symlinks to `libmpi.so.20.10.1`. Btw, this is a real library name from OpenMPI.